### PR TITLE
feat: add zapper support

### LINKED ISSUE #6: Add zapper support
there is already a nwc pool that subscribes to payment events for all lightning addresses. We need to act on the zap events to write the zap receipt (maybe https://github.com/getAlby/lite/issues/3 needs to be implemented first)

### PR DESCRIPTION
**Pull request by im-adithya**
Fixes #6

Adds zapper

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -11,3 +11,7 @@ if (!databaseUrl) {
   Deno.exit(1);
 }
 export const DATABASE_URL = databaseUrl;
+
+export const NOSTR_NIP57_PRIVATE_KEY = Deno.env.get("NOSTR_NIP57_PRIVATE_KEY") || "";
```
Comment (rolznz) [inline src/constants.ts:15]: Can you update the .env.example and update the README with instructions for people to know what to do?

File: src/nwc/nwcPool.ts [hunk 2]
```diff
@@ -24,13 +35,71 @@ export class NWCPool {
...
+    });
+    
+    if (!response.ok) {
+      logger.error("failed to publish zap", {
+        user_id: userId,
+        event_id: signedEvent.id, 
+        payment_hash: transaction.payment_hash, 
+        relays,
+        response_status: response.status,
+      });
```
Comment (rolznz) [inline src/nwc/nwcPool.ts:95]: there should be a return here

File: src/constants.ts (hunk 1 above)
Comment (im-adithya) [reply src/constants.ts:15]: Done!

File: src/nwc/nwcPool.ts (hunk 2 above)
Comment (im-adithya) [reply src/nwc/nwcPool.ts:95]: Ah thanks, fixed!

File: src/nwc/nwcPool.ts [hunk 3]
```diff
@@ -24,13 +33,73 @@ export class NWCPool {
...
+      return;
+    }
+
+    const zapReceipt = makeZapReceipt({
+      zapRequest: JSON.stringify(requestEvent),
+      preimage: transaction.preimage,
+      bolt11: transaction.invoice,
+      paidAt: new Date(transaction.settled_at * 1000)
+    })
+    const relays = requestEvent.tags.filter(tag => tag[0] === 'relays')[0].slice(1);
```
Comment (rolznz) [inline src/nwc/nwcPool.ts:66]: You could use `find` rather than `filter` since you just take the first element

File: src/nwc/nwcPool.ts [hunk 4]
```diff
@@ -24,13 +33,73 @@ export class NWCPool {
...
+      } else {
+        failedRelays.push(relay);
+      }
+    });
+
+    if (failedRelays.length === relays.length) {
+      logger.error("failed to publish zap", {
+        user_id: userId,
+        event_id: signedEvent.id, 
+        payment_hash: transaction.payment_hash,
```
Comment (rolznz) [inline src/nwc/nwcPool.ts:92]: should we include the `failed_relays` here too?

**Review by rolznz — APPROVED** [review]
utACK

**Pull request merged**